e2eshark/tools/stubs/turbinemodel.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2024 Advanced Micro Devices
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This file constitutes end  part of runmodel.py
+# this is appended to the model.py in test dir
+
+import argparse
+import pickle
+
+from turbine_models.model_builder import HFTransformerBuilder
+import shark_turbine.aot as aot
+from commonutils import getOutputTensorList, E2ESHARK_CHECK_DEF, postProcess
+
+msg = "The script to run a model test"
+parser = argparse.ArgumentParser(description=msg, epilog="")
+
+parser.add_argument(
+    "-d",
+    "--todtype",
+    choices=["default", "fp32", "fp16", "bf16"],
+    default="default",
+    help="If not default, casts model and input to given data type if framework supports model.to(dtype) and tensor.to(dtype)",
+)
```

Comment:
Does `model.to(dtype)` change the entire model...? How does that even work? (Lots of "real" workloads are expected to only reduce precision in a few specific places)